### ISSUE TITLE
Everytime i close session i gets cleared.

### ISSUE THREAD

**Issue by Saifallak** [wppconnect-team/wppconnect-server#1170]
## Description

Everytime i close session i gets cleared.

## Environment

- **Wppconnect-server version(s):** 1.8.5
- **Browser:** Chromium
- **OS:** Centos 7
- **Node version:** 14
- **WhatsApp version:** 2.2307.x
- **MultiDevice (BETA):** yes
- **config.json:** defualt.

## Steps to Reproduce

1. Start session
2. close it
3. asks for qrcode.

## Log Output

```
verbose:  [984f8438-8e31-401e-93ad-42329e677740:page] Page Closed
info: 2023-03-31T15:13:07.730Z browserClose


info:     [984f8438-8e31-401e-93ad-42329e677740:browser] Using browser folder './userDataDir/984f8438-8e31-401e-93ad-42329e677740'
info:     [984f8438-8e31-401e-93ad-42329e677740:browser] Initializing browser...
http:     [984f8438-8e31-401e-93ad-42329e677740:browser] checking headless...
http:     [984f8438-8e31-401e-93ad-42329e677740:browser] headless option is active, browser hidden
info:     [984f8438-8e31-401e-93ad-42329e677740:client] Initializing...
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] Setting WhatsApp WEB version to 2.2307.x
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Waiting page load
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] Loading WhatsApp WEB
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] WhatsApp WEB loaded
verbose:  [984f8438-8e31-401e-93ad-42329e677740:page] Page loaded
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] Injecting wapi.js
Error: Execution context was destroyed, most likely because of a navigation.
    at rewriteError (............/public_html/node_modules/puppeteer-core/src/common/ExecutionContext.ts:391:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at ExecutionContext._ExecutionContext_evaluate (............/public_html/node_modules/puppeteer-core/src/common/ExecutionContext.ts:276:56)
    at ExecutionContext.convertArgument (............/public_html/node_modules/puppeteer-core/src/common/ExecutionContext.ts:332:15)
    at async Promise.all (index 0)
    at ExecutionContext._ExecutionContext_evaluate (............/public_html/node_modules/puppeteer-core/src/common/ExecutionContext.ts:302:20)
    at Frame.addScriptTag (............/public_html/node_modules/puppeteer-core/src/common/Frame.ts:814:7)
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] wapi.js failed
verbose:  [984f8438-8e31-401e-93ad-42329e677740:page] Page loaded
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] Injecting wapi.js
info:     [984f8438-8e31-401e-93ad-42329e677740:session] Session Unpaired
verbose:  [984f8438-8e31-401e-93ad-42329e677740:client] wapi.js injected
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Connected
info: 2023-03-31T15:13:20.727Z inChat


debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onMessage function
info:     [984f8438-8e31-401e-93ad-42329e677740:client] WhatsApp WEB version: 2.2313.8
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onAnyMessage function
info:     [984f8438-8e31-401e-93ad-42329e677740:client] WA-JS version: 2.22.2
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onAck function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onNotificationMessage function
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Checking is logged...
info:     [984f8438-8e31-401e-93ad-42329e677740:client] Auto close configured to 60s
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Waiting for QRCode Scan...
info: 2023-03-31T15:13:20.977Z notLogged


debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onParticipantsChanged function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onStateChange function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onStreamChange function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onIncomingCall function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onInterfaceChange function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onPresenceChanged function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onLiveLocation function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onAddedToGroup function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onRevokedMessage function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onReactionMessage function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Exposing onPollResponse function
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Emitting onInterfaceChange event (1 registered)
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Current state: QR (CONNECTING)
info:     [984f8438-8e31-401e-93ad-42329e677740:browser] Using browser folder './userDataDir/984f8438-8e31-401e-93ad-42329e677740'
info:     [984f8438-8e31-401e-93ad-42329e677740:browser] Initializing browser...
debug:    [984f8438-8e31-401e-93ad-42329e677740:client] Emitting onInterfaceChange event (1 registered)
http:     [984f8438-8e31-401e-93ad-42329e677740:client] Current state: QR (CONNECTING)
warn: 2023-03-31T15:13:21.559Z Error calling Webhook. Request failed with status code 500 - AxiosError: Request failed with status code 500
    at settle (............/public_html/node_modules/axios/lib/core/settle.js:19:12)
    at Unzip.handleStreamEnd (............/public_html/node_modules/axios/lib/adapters/http.js:556:11)
    at Unzip.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
info:     [984f8438-8e31-401e-93ad-42329e677740:client] Waiting for QRCode Scan (Attempt 1)...:
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▀▀█ ▀ ▄█ ▀▄▄▄  ▄▄  █ ▄  ▄█▄▄ ▀ ▀ █ ▀██ ▄▄▄▄▄ █
 █▀█▀▄█ ███▀  ▄  ▀██ █▄▄▄█ █▀█▀█ ██ ▄▀ ▀ ▄▄▄█▄▄▄ ▀▀█▀▀▄▄  ▄▀▄ ███▄▄█▄██ ██
█▄▄▄▄▄▄▄█▄▄██▄██▄█▄▄█▄▄█▄▄▄▄▄█▄▄▄▄▄██▄█▄▄██▄▄█▄▄██▄█▄██
 {"code":"2@KfJvbA4Ib2XHcSEyurwz5nw9BHxbl8BTmP05pB2e2snAeUm7jyXcCBSUP4NEvwLvI55gAdkChMV4Ug==,BRfC2E5VaILfp2DU39GvYOGCkwVjRFyr3PV1FpI8Rhc=,q1bl9y+kXOmhNyzMpKAJJ8ewlayxx18DWzkD1WCZgSA=,DgVWikKGT9iqjYD7AuvXyx2hIIYbbPIvzNtCFJxCAGU="}
http:     [984f8438-8e31-401e-93ad-42329e677740:browser] checking headless...
```

**Comment by Saifallak** [comment]
Okay after some more testing.

I found that the issue happens by the following:

- Start session normally.
- Scan QRCode.
- turn session OFF (aka. Close Session)

- Start the same session
- fetch session status immediately after its started.
- it will unpair the session

i guess this happens because as soon as i ask server for session status before it fully loads it assumes it's crashed or something like that. so it makes a new session.
and in logs above as you see, it starts and in-chat then unpairs.

**Comment by icleitoncosta** [comment]
Hello, how is a version of @wppconnect-team/wppconnect?

**Comment by Saifallak** [comment]
```
"@wppconnect-team/wppconnect@^1.23.0":
  version "1.23.0"
  resolved "https://registry.yarnpkg.com/@wppconnect-team/wppconnect/-/wppconnect-1.23.0.tgz#ac4ec3c103eb261d011f09a79f2ea91152e46f99"
  integrity sha512-thqHkqxw0smYsWU4BSCbwAdWztiNbikmhg/Il+Lzz4zMTtp4s48p18J0DF77RG2i2dZvSPUqt6VVbdBMrPo6eQ==
  dependencies:
    "@wppconnect/wa-js" "^2.22.0"
    "@wppconnect/wa-version" "^1.2.5"
    atob "^2.1.2"
    axios "^1.3.4"
    boxen "^5.1.2"
    catch-exit "^1.2.2"
    chalk "~4.1.2"
    chrome-launcher "^0.15.1"
    execa "^5.1.1"
    file-type "~16.5.4"
    futoin-hkdf "^1.5.1"
    latest-version "^5.1.0"
    logform "^2.5.1"
    lookpath "^1.2.2"
    mime-types "^2.1.35"
    puppeteer "^19.7.3"
    puppeteer-extra "^3.3.6"
    puppeteer-extra-plugin-stealth "^2.11.2"
    puppeteer-extra-plugin-user-data-dir "^2.4.1"
    puppeteer-extra-plugin-user-preferences "^2.4.1"
    qrcode-terminal "^0.12.0"
    reflect-metadata "^0.1.13"
    rimraf "^3.0.2"
    sanitize-filename "^1.6.3"
    sharp "^0.31.3"
    tmp "^0.2.1"
    tree-kill "^1.2.2"
    winston "^3.8.2"
    ws "^8.12.1"
  optionalDependencies:
    fsevents "^2.3.2"

"@wppconnect/wa-js@^2.22.0":
  version "2.22.0"
  resolved "https://registry.yarnpkg.com/@wppconnect/wa-js/-/wa-js-2.22.0.tgz#a3845fc10bd3a30b4cb4da353422bfed2aa6736b"
  integrity sha512-TWwVkYEyG+IftPCQbKag7nuN0dE30pp2jmVIjs5KRW1R1Eq9X85qy0KJpMW9teWlKzTU0awFQvHn/t0L6k8iXw==

"@wppconnect/wa-version@^1.2.5":
  version "1.2.5"
  resolved "https://registry.yarnpkg.com/@wppconnect/wa-version/-/wa-version-1.2.5.tgz#8a4bc78f24d917a1e72fd1bd469182e93aa39aac"
  integrity sha512-Fxg+CirLQn7fPwi43E7LtnLeVGbltGPtkzB/snwJ2dKWYL5CG5JM6QMRPlKKFBEO1PVI+PVcxHF0QLxmWb5NGg==
  dependencies:
    node-fetch "^2.6.9"
    semver "^7.3.8"

```

**Comment by Saifallak** [comment]
this is snippet from my php code that calls the server

```php
        $server = $this->getServer();
        $session = $server->fetchSession(); // /api/{WPP-SESSION-ID}/status-session

        if ($session['status'] === 'CLOSED') {
            $server->startSession(true); // /api/{WPP-SESSION-ID}/start-session with waitQrCode = true
        }

        if (in_array(Str::upper($session['status'] ?? ''), ['CONNECTED', 'QRREADSUCCESS', 'ISLOGGED', 'INCHAT'])) {
                $this->host = $server->getHostDevice()['response'] ?? []; // /api/{WPP-SESSION-ID}/host-device
        }
```

**Comment by Saifallak** [comment]
i guess here that i'm starting the session then getting host device immediately,
i will do more tests anyway tomorrow.

**Comment by fabioselau077** [comment]
Resolved? Same problem
#1331

**Comment by Saifallak** [comment]
> Resolved? Same problem #1331

I don't know, i didn't test it, little busy to debug, i just made a little timer before sending any request to the session.
